src/lua/init.c

```diff
@@ -930,13 +930,20 @@ run_script_f(va_list ap)
 		is_a_tty = inj->iparam;
 	});
 
+	const char *env_background = getenv("TT_BACKGROUND");
+	bool is_background = false;
+	if (env_background) {
+		is_background = (strcmp(env_background, "true") == 0 ? true : false);
+	}
```

Comment:
I'm sorry if I'm too early, but the pull request is not marked as a draft and I guess I can raise a early discussion here. Don't take it as a nit, please.

If we'll do `tarantool -e "box.cfg{background = true, listen = 3301, log = 'tarantool.log', pid_file = 'tarantool.pid'}"`, we'll get the same error in the log file without any `TT_*` environment variable.

We should access the option via `cfg_getb()` from `src/cfg.h` if the solution is to skip stdin reading when we're going to background in the `-e <...>` script.

However I would think here a bit more. What if `-l fiber -e 'fiber.sleep(1) box.cfg{<...>}'` is passed?

Reply:
Both variants looks working now.